.github/workflows/helm-chart.yml

```diff
@@ -22,35 +25,52 @@ jobs:
 
       - name: Configure git
         run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
 
       - name: Install helm
         uses: azure/setup-helm@v4
         env:
           GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
 
+      - name: Install helm-docs
+        env:
+          HELM_DOCS_VERSION: "1.14.2"
+        run: |
+          cd /tmp
+          wget https://github.com/norwoodj/helm-docs/releases/download/v"${HELM_DOCS_VERSION}"/helm-docs_"${HELM_DOCS_VERSION}"_Linux_x86_64.tar.gz
+          tar -xvf helm-docs_"${HELM_DOCS_VERSION}"_Linux_x86_64.tar.gz
+          sudo mv helm-docs /usr/local/bin
+
       - name: Set helm chart appVersion from Version.props
         run: |
           export VERSION_PROPS=$(awk -F'[<>]' '/VersionPrefix/{print $3}' Version.props)
           sed -i -e 's#Version.props#"'${VERSION_PROPS}'"#g' charts/garnet/Chart.yaml
 
-      - name: Helm lint and package
+      - name: Helm lint, helm-docs and helm package
         run: |
           mkdir .cr-release-packages
           for chart in $(find charts -mindepth 1 -maxdepth 1 -type d); do
             if [ -z "${chart:-}" ]; then
               break
             fi
             helm lint "${chart}"
+            helm-docs --document-dependency-values --chart-search-root "${chart}"
             helm package "${chart}" --dependency-update --destination .cr-release-packages
           done
 
+      - name: Git commit charts/garnet/README.md and push to main
+        run: |
+          git checkout -- charts/garnet/Chart.yaml
+          git add charts/garnet/README.md
+          git commit -s -m "Apply README.md changes in helm chart"
```

Comment:
What if there were no changes to the file?

Reply:
This action is triggered by changes to the helm chart files so there will be at least a `version` change and therefore helm-docs will update the README.md file.

Reply:
Need to test that if someone just changes something minor like formatting in the chats folder, that will end the CI silently. The CI triggers for a change to any file in the charts folder, so.

Reply:
With commit 0fd91592e447da5646ebcd95e8dce55a570fbf79 , we only trigger the github workflow on a commit on `charts/garnet/Chart.yaml`.
* If the `version` is updated, a new helm chart package will be built/pushed and a new PR is created to update the README.md.
* If only another thing is updated (ie. contributors, icon, home ...), the allready existing helm-chart with the same `version` is overwrited and a new PR is created to update the README.md.